source/js/buyers-guide/bg-main.js

```diff
@@ -16,64 +17,81 @@ import ProductGA from "./product-analytics.js";
-            ReactDOM.render(<Filter whenLoaded={() => resolve()} />, filter);
-          })
-        );
+        if (filter) {
+          apps.push(
+            new Promise(resolve => {
+              ReactDOM.render(<Filter whenLoaded={() => resolve()} />, filter);
+            })
+          );
+        }
       }
-    }
 
-    if (document.getElementById(`pni-product-page`)) {
-      ProductGA.init();
-
-      // Set up help text accordions where necessary:
-      let productBox = document.querySelector(`.product-detail .h1-heading`);
-      let productName = productBox ? productBox.textContent : `unknown product`;
-      let criteriaWithHelp = document.querySelectorAll(
-        `.criterion button.toggle`
-      );
-
-      if (criteriaWithHelp.length > 0) {
-        Array.from(criteriaWithHelp).forEach(button => {
-          let help = button.closest(`.criterion`).querySelector(`.helptext`);
-
-          button.addEventListener(`click`, () => {
-            button.classList.toggle(`open`);
-            help.classList.toggle(`open`);
-
-            if (help.classList.contains(`open`)) {
-              ReactGA.event({
-                category: `product`,
-                action: `expand accordion tap`,
-                label: `detail view on ${productName}`
-              });
-            }
+      if (document.getElementById(`pni-product-page`)) {
+        ProductGA.init();
+
+        // Set up help text accordions where necessary:
+        let productBox = document.querySelector(`.product-detail .h1-heading`);
+        let productName = productBox
+          ? productBox.textContent
+          : `unknown product`;
+        let criteriaWithHelp = document.querySelectorAll(
+          `.criterion button.toggle`
+        );
+
+        if (criteriaWithHelp.length > 0) {
+          Array.from(criteriaWithHelp).forEach(button => {
+            let help = button.closest(`.criterion`).querySelector(`.helptext`);
+
+            button.addEventListener(`click`, () => {
+              button.classList.toggle(`open`);
+              help.classList.toggle(`open`);
+
+              if (help.classList.contains(`open`)) {
+                ReactGA.event({
+                  category: `product`,
+                  action: `expand accordion tap`,
+                  label: `detail view on ${productName}`
+                });
+              }
+            });
           });
-        });
+        }
       }
-    }
 
-    // Record that we're done, when we're really done.
-    Promise.all(apps).then(() => {
-      window[`bg-main-js:react:finished`] = true;
+      // Record that we're done, when we're really done.
+      Promise.all(apps).then(() => {
+        window[`bg-main-js:react:finished`] = true;
+      });
+    });
+  },
+
+  fetchEnv(callback) {
+    let envReq = new XMLHttpRequest();
+
+    envReq.addEventListener(`load`, () => {
+      callback.call(this, JSON.parse(envReq.response));
     });
+
+    envReq.open(`GET`, `/environment.json`);
```

Comment:
This returns `{"PULSE_API_DOMAIN": "https://network-pulse-api-production.herokuapp.com", "PULSE_DOMAIN": "www.mozillapulse.org", "NETWORK_SITE_URL": "", "TARGET_DOMAINS": null}`, and that last one looks very wrong. Is there a way for us to make that be `[]` instead of `null`?

Reply:
PR updated to address this!